ingest/src/main/java/com/connexta/ingest/service/impl/IngestServiceImpl.java

```diff
@@ -9,30 +9,60 @@
 import com.connexta.ingest.adaptors.S3Adaptor;
 import com.connexta.ingest.service.api.IngestRequest;
 import com.connexta.ingest.service.api.IngestService;
+import com.connexta.ingest.transform.TransformClient;
+import com.connexta.transformation.rest.models.TransformRequest;
+import com.connexta.transformation.rest.models.TransformResponse;
+import java.io.IOException;
+import java.net.URL;
 import java.util.UUID;
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.Resource;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.web.multipart.MultipartFile;
 
 @Service
+@AllArgsConstructor
 public class IngestServiceImpl implements IngestService {
-
+  private static final Logger LOGGER = LoggerFactory.getLogger(IngestServiceImpl.class);
   private final S3Adaptor s3Adaptor;
-
-  public IngestServiceImpl(S3Adaptor s3Adaptor) {
-    this.s3Adaptor = s3Adaptor;
-  }
+  private final TransformClient transformClient;
 
   @Override
-  public UUID ingest(
+  public void ingest(
       String acceptVersion,
       Long fileSize,
       String mimeType,
       MultipartFile file,
       String title,
-      String fileName) {
+      String fileName)
+      throws IOException {
+    final String ingestId = UUID.randomUUID().toString().replace("-", "");
+    s3Adaptor.upload(
+        new IngestRequest(acceptVersion, fileSize, mimeType, file, title, fileName), ingestId);
+
+    final String url = new URL("TODO/retrieve/" + ingestId).toString();
```

Comment:
✏️ This URL should come from a configuration file

Reply:
Yep. This is a know issue. We will address this in a future PR.